Upstream-Author: Christos Zoulas <christos@zoulas.com>
Date: Fri Feb 21 14:32:48 2014 +0000
Upstream-Commit: c0c0032b9e9eb57b91fefef905a3b018bab492d9
Description:
 Fix memory leak (Anatol Belski)

--- a/src/softmagic.c
+++ b/src/softmagic.c
@@ -1755,12 +1755,16 @@
 		ms->offset = soffset;
 		if (rv == 1) {
 			if ((ms->flags & (MAGIC_MIME|MAGIC_APPLE)) == 0 &&
-			    file_printf(ms, F(m->desc, "%u"), offset) == -1)
+			    file_printf(ms, F(m->desc, "%u"), offset) == -1) {
+				free(rbuf);
 				return -1;
-			if (file_printf(ms, "%s", rbuf) == -1)
+			}
+			if (file_printf(ms, "%s", rbuf) == -1) {
+				free(rbuf);
 				return -1;
-			free(rbuf);
+			}
 		}
+		free(rbuf);
 		return rv;
 
 	case FILE_USE:
